src/v/kafka/protocol/kafka_batch_adapter.cc

```diff
@@ -77,6 +79,7 @@ model::record_batch_header kafka_batch_adapter::read_header(iobuf_parser& in) {
       .producer_epoch = producer_epoch,
       .base_sequence = base_sequence,
       .record_count = record_count};
+    header.ctx.term = model::term_id(partition_leader_epoch);
```

Comment:
hmm I think `header.ctx.term` is some internal thing. what is failing? is it crc check or something?

Reply:
no, we just need to have exactly the same binary implementation when read/written

Reply:
this isn't symmetric with the changes to the response_writer. and that is what is confusing here currently.